signer/src/testing/block_observer.rs

```diff
@@ -355,17 +354,21 @@ impl StacksInteract for TestHarness {
             .cloned()
             .ok_or(Error::MissingBlock)
     }
-    async fn check_pre_nakamoto_block(&self, _: &StacksBlockHash) -> Result<(), Error> {
-        unimplemented!()
-    }
+
     async fn get_tenure_headers(
         &self,
-        block_id: &StacksBlockHash,
+        burnchain_block_height: BitcoinBlockHeight,
     ) -> Result<TenureBlockHeaders, Error> {
+        let burnchain_hash = self
+            .bitcoin_blocks
+            .iter()
+            .find(|block| block.height == burnchain_block_height)
+            .ok_or(Error::MissingBlock)?
+            .block_hash;
         let (stx_block_id, stx_block, btc_block_id) = self
             .stacks_blocks
             .iter()
-            .find(|(id, _, _)| id == &block_id.into())
+            .find(|(_, _, btc_hash)| &burnchain_hash == btc_hash)
             .ok_or(Error::MissingBlock)?;
```

Comment:
nit: once we have `burnchain_hash` we can just use that in the `skip_while`/`take_while` below, we don't need to compute `stx_block_id` & co